InterfaceMATLAB/tsgWriteMatrix.m

```diff
@@ -13,28 +13,56 @@ function tsgWriteMatrix(filename, mat)
 % 5 6 7 8
 % 9 10 11 12
 %
+%
+% the complex matrix [1+2i 3+4i; 5+6i 7+8i; 9+10i 11+12i] is written as
+%
+% 3 2
+% 1 2 3 4
+% 5 6 7 8
+% 9 10 11 12
+%
+
+bIsComplex = ~isreal(mat);
 
 if (prod(size(mat)) < 1000) % small matrix, use ascii format
 
     fid = fopen(filename, 'w');
 
-    fprintf(fid, '%d  %d\n', size(mat, 1), size(mat, 2)); % load the number of points
```

Comment:
* same as `tsgReadMatrix()`, this is only ever called on `setHierarchicalCoefficients()`, in that function you should first make sure that complex numbers are used only when the grid is `fourier` and Global or Sequence. You should also convert the matrix to reals and pass only a single double matrix to `tsgWriteMatrix()`
```
real_matrix = [complex_matrix, complex_matrix]; % defines the dimension
real_matrix(:, 1:2:end) = real(complex_matrix);
real_matrix(:, 2:2:end) = imag(complex_matrix);
```